hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/tokentypes/Token.java

```diff
@@ -0,0 +1,244 @@
+                    customFeesFromMod(token.customFees()),
+                    keyFromMod(token.adminKey()),
+                    keyFromMod(token.feeScheduleKey()),
+                    keyFromMod(token.freezeKey()),
+                    keyFromMod(token.kycKey()),
+                    keyFromMod(token.pauseKey()),
+                    keyFromMod(token.supplyKey()),
+                    keyFromMod(token.wipeKey()));
+        } catch (InvalidKeyException e) {
+            // todo check how to handle invalid key exceptions
+            throw new RuntimeException(e);
+        }
+
+        Objects.requireNonNull(tokenRes.tokenType, "tokenType");
+        Objects.requireNonNull(tokenRes.tokenSupplyType, "tokenSupplyType");
+        Objects.requireNonNull(tokenRes.symbol, "symbol");
+        Objects.requireNonNull(tokenRes.name, "name");
+        Objects.requireNonNull(tokenRes.memo, "memo");
+        Objects.requireNonNull(tokenRes.adminKey, "adminKey");
+        Objects.requireNonNull(tokenRes.feeScheduleKey, "feeScheduleKey");
+        Objects.requireNonNull(tokenRes.freezeKey, "freezeKey");
+        Objects.requireNonNull(tokenRes.kycKey, "kycKey");
+        Objects.requireNonNull(tokenRes.pauseKey, "pauseKey");
+        Objects.requireNonNull(tokenRes.supplyKey, "supplyKey");
+        Objects.requireNonNull(tokenRes.wipeKey, "wipeKey");
+
+        return tokenRes;
+    }
+
+    private static EntityId idFromMod(@Nullable final AccountID accountId) {
+        return null == accountId ? EntityId.MISSING_ENTITY_ID : new EntityId(0L, 0L, accountId.accountNumOrThrow());
+    }
+
+    private static List<FcCustomFee> customFeesFromMod(List<CustomFee> customFees) {
+        List<FcCustomFee> fcCustomFees = new ArrayList<>();
+        customFees.stream().forEach(fee -> {
+            var fcCustomFee = FcCustomFee.fromGrpc(PbjConverter.fromPbj(fee));
+            fcCustomFees.add(fcCustomFee);
+        });
+        return fcCustomFees;
+    }
+
+    static TokenSupplyType supplyTypeFromMono(
+            @NonNull com.hedera.node.app.service.mono.state.enums.TokenSupplyType tokenSupplyType) {
+        return (tokenSupplyType.equals(com.hedera.node.app.service.mono.state.enums.TokenSupplyType.INFINITE))
+                ? TokenSupplyType.INFINITE
+                : TokenSupplyType.FINITE;
+    }
+
+    private static Optional<JKey> keyFromMod(@Nullable Key key) throws InvalidKeyException {
+        return key == null ? Optional.empty() : Optional.ofNullable(JKey.mapKey(key));
+    }
+
+    @NonNull
+    String getKeyProfile() {
+        final var adminKeyOk = jkeyPresentAndOk(adminKey);
+
+        return getKeyDescription((c, ojk) -> {
+            if (!jkeyPresentAndOk(ojk)) return "    ";
+            if (!adminKeyOk) return c + "   ";
+            if (c == 'A') return "A   ";
+            if (jkeyDeepEqualsButBothNullIsFalse(ojk.get(), adminKey.get())) return c + "=A ";
+            return c + "   ";
+        });
+    }
```

Comment:
It would be nice if these `getKeyXXX()` were consistent across entity types, if possible (e.g. dumping keys on accounts or keys on topics would dump the same format)

Reply:
Actually, the keys are dumped by `ThingsToStrings::toStringOfJKey` and all the dumpers I checked use this method too. 
The `getKeyXXX()` methods, are used by [these](https://github.com/hashgraph/hedera-services/pull/11543/files#diff-c4e227a3be393f13d5c3c67b28d2548a84707f74ef88a21609960c670b88e9e3R175-R176) additional reports (I'm not sure if we need them here. I was following `DumpTokensSubcommand` logic).

Reply:
ok, thanks Jivko and David!